src/include/snapper/dialogs.rb

```diff
@@ -47,198 +44,409 @@ def initialize_snapper_dialogs(include_target)
-          modified = Snapper.ModifySnapshot(args)
+      created = Snapper.CreateSnapshot(args) if ret == :ok
+      created
+    end
+
+    # Popup for deleting existing snapshot
+    # @return true if snapshot was deleted
+    def DeleteSnapshotPopup(snapshots)
+      snaps = []
+      post_snaps = []
+      snapshots.each do |s|
+        if !s.post?
+          snaps << s.number
+        else
+          post_snaps << { pre_num: s.pre.number, num: s.number }
         end
-        if type == :POST && snapshot_modified(pre_snapshot, pre_args)
-          modified = Snapper.ModifySnapshot(pre_args) || modified
+      end
+
+      nums = snaps + post_snaps.map { |k| "#{k[:pre_num]}&#{k[:num]}" }
+
+      # yes/no popup question
+      return false if !Popup.YesNo(_("You have selected those snapshots: %{num}") % { num: nums.join(",") })
+
+      Snapper.DeleteSnapshot(snaps)
+      post_snaps.each do |k|
+        Snapper.DeleteSnapshot([k[:pre_num], k[:num]])
+      end
+      true
+    end
+
+    # Summary dialog
+    # @return dialog result
+    def SummaryDialog
+      # summary dialog caption
+      caption = _("Snapshots")
+
+      # update list of snapshots
+      Wizard.SetContentsButtons(
+        caption,
+        snapshots_table,
+        Ops.get_string(@HELPS, "summary", ""),
+        Label.BackButton,
+        Label.CloseButton
+      )
+      Wizard.HideBackButton
+      Wizard.HideAbortButton
+
+      UI.SetFocus(Id(:snapshots_table))
+
+      refresh_buttons
+      summary_event_loop
+    end
+
+    def format_diff(diff, textmode)
+      lines = String.EscapeTags(diff).split("\n")
+      return lines.join("<br>") if textmode
+      # colorize diff output
+      lines.map! do |line|
+        case line[0]
+        when "+"
+          line = "<font color=blue>#{line}</font>"
+        when "-"
+          line = "<font color=red>#{line}</font>"
         end
+        line
       end
 
-      modified
+      # show fixed font in diff
+      "<pre>#{lines.join("<br>")}</pre>"
     end
 
+    def textmode?
+      display_info = UI.GetDisplayInfo
+      display_info["TextMode"] || false
+    end
 
-    # Popup for creating new snapshot
-    # @return true if new snapshot was created
-    def CreateSnapshotPopup(pre_snapshots)
-      pre_snapshots = deep_copy(pre_snapshots)
-      created = false
-      pre_items = Builtins.maplist(pre_snapshots) do |s|
-        Item(Id(s), Builtins.tostring(s))
+    # @return dialog result
+    def ShowDialog
+      # dialog caption
+      caption = _("Selected Snapshot Overview")
+
+      current_file = nil
+
+      snapshot = Snapper.selected_snapshot
+
+      from = snapshot
+      to = snapshot.pre? ? snapshot.post : nil
+      if snapshot.post?
+        from = snapshot.pre
+        to = snapshot
+      end
+
+      # busy popup message
+      Popup.ShowFeedback("", _("Calculating changed files..."))
+      files_tree = Snapper.ReadModifiedFilesTree(from, to)
+      Popup.ClearFeedback()
```

Comment:
better to use new feedback with block  https://github.com/yast/yast-yast2/blob/8eb9214265fdf8afcae1531c5a0309799028d79d/library/general/src/modules/Popup.rb#L1071